app/controllers/trains_controller.rb

```diff
@@ -1,15 +1,36 @@
 class TrainsController < ApplicationController
+  helper_method :train
+
   def index
     @trains = TrainApi.new.index
   end
 
+  def edit
+    flash["errors"] = nil
+    @train = TrainApi.new.show(params[:id])
+  end
+
   def create
   end
 
-  def edit
+  def update
+    train_params = params.require(:train).permit(:name, :train_model_name, :number_of_cars, :max_weight_capacity, :active)
```

Comment:
normally we do params in a before action, as you will need this exact same code for the create action too